airflow/ti_deps/deps/pool_slots_available_dep.py

```diff
@@ -41,8 +43,7 @@ def _get_dep_statuses(self, ti, session, dep_context=None):
         from airflow.models.pool import Pool  # To avoid a circular dependency
 
         pool_name = ti.pool
-
-        pools = session.query(Pool).filter(Pool.pool == pool_name).all()
+        pools = session.scalars(select(Pool).where(Pool.pool == pool_name)).all()
```

Comment:
This can be rewritten as

```python
pool = session.scalar(select(Pool).where(Pool.pool == pool_name))
if pool is None:
    yield ...
    return
# Replace "pools[0]" with "pool"...